syntax = "proto3";
package eos.ns;

//------------------------------------------------------------------------------
// File metadata protocol buffer object
//------------------------------------------------------------------------------
message FileMdProto {
  uint64 id = 1;
  uint64 cont_id = 2;
  uint64 uid = 3;
  uint64 gid = 4;
  uint64 size = 5;
  uint32 layout_id = 6;
  uint32 flags = 7;
  bytes name = 8;
  bytes link_name = 9;
  bytes ctime = 10; // change time
  bytes mtime = 11; // modification time
  bytes checksum = 12;
  repeated uint32 locations = 13;
  repeated uint32 unlink_locations = 14;
  map<string, bytes> xattrs = 15;
  bytes stime = 16; // server sync time
  bytes atime = 17; // server access time
  map<uint32, bytes> altchecksums = 18; // alternative checksums
  uint64 cloneid = 256;     // transient
  bytes clonefst = 257;     // transient
}
